app/models/user.rb

```diff
@@ -6,7 +6,68 @@ class User < ApplicationRecord
 
   validates :name, presence: true, length: { maximum: 20 }
 
+  # rubocop:disable Layout/LineLength:
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :recieved_friendship, class_name: 'Friendship', foreign_key: 'reciever_user_id', inverse_of: :reciever_user, dependent: :destroy
+  has_many :requested_friendship, class_name: 'Friendship', foreign_key: 'requester_user_id', inverse_of: :requester_user, dependent: :destroy
+  # rubocop:enable Layout/LineLength:
+  def friends
+    friends = []
+    Friendship.all.each do |request|
+      if request.status == true
+        if request.requester_user_id == id
+          friends << User.find(request.reciever_user_id)
+        elsif request.reciever_user_id == id
+          friends << User.find(request.requester_user_id)
+        end
+      end
+    end
+    friends
+  end
+
+  def pending_requests
+    pending_friend = []
+    Friendship.all.each do |pending|
+      next unless pending.status.nil?
+
+      pending_friend << pending if pending.reciever_user_id == id
+    end
+    pending_friend
+  end
+
+  def friend_requests
+    requests = []
+    Friendship.all.each do |request|
+      next unless request.status.nil?
+
+      requests << request if request.requester_user_id == id
+    end
+    requests
+  end
```

Comment:
You already have the association `requested_friendship` then why don't you use it? 
**HINT**: `user.requested_friendships` And you can select only the ones with `status: false`